tester/plugin/pytest_interface_tester.py

```diff
@@ -0,0 +1,412 @@
+import logging
+import operator
+import tempfile
+from pathlib import Path
+from subprocess import PIPE, Popen
+from typing import (
+    TYPE_CHECKING,
+    Callable,
+    Iterable,
+    Literal,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+)
+
+import pytest
+from ops.testing import CharmType
+from scenario.scenario import Scenario
+from scenario.structs import (
+    CharmSpec,
+    Event,
+    EventMeta,
+    RelationMeta,
+    Scene,
+    State,
+    event,
+    relation,
+)
+
+from collect_interface_tests import InterfaceTestSpec, gather_test_spec_for_version
+
+if TYPE_CHECKING:
+    from tester.plugin.interface_test import DataBagSchema, InterfaceTestCase
+
+Callback = Callable[[Scene], None]
+
+logger = logging.getLogger("pytest_interface_tester")
+ROLE_TO_ROLE_META = {"provider": "provides", "requirer": "requires"}
+
+
+class InterfaceTester:
+    def __init__(
+        self,
+        repo: str = "https://github.com/PietroPasotti/charm-relation-interfaces",
+        branch: str = "main",
+        base_path: str = "interfaces",
+    ):
+        self._repo = repo
+        self._branch = branch
+        self._base_path = base_path
+        self._target = None
+        self._interface_name = None
+        self._state_template = None
+
+    def configure(
```

Comment:
Why this `configure` in general? Can't we leverage `__init__`?

Reply:
the idea is to make `interface_tester` a pytest fixture, which means we need to initialize the object at a time where we don't know what interface we will be testing.